main.go

```diff
@@ -219,8 +242,10 @@ func watchTriggers() {
 		currQuote, err := types.ParseQuote(string(d.Body[:]))
 		failOnError(err, "Failed to parse Quote")
 		consoleLog.Debugf("New Quote for %s at price %s\n", currQuote.Stock, currQuote.Price.String())
-		buyTree, buyFound := autoTxStore[TreeKey{currQuote.Stock, "Buy"}] // tree, found
-		sellTree, sellFound := autoTxStore[TreeKey{currQuote.Stock, "Sell"}]
+		AutoTxStorage.Mutex.Lock()
+		buyTree, buyFound := AutoTxStorage.AutoTxTrees[TreeKey{currQuote.Stock, "Buy"}] // tree, found
+		sellTree, sellFound := AutoTxStorage.AutoTxTrees[TreeKey{currQuote.Stock, "Sell"}]
+		AutoTxStorage.Mutex.Unlock()
 		if !buyFound && !sellFound {
```

Comment:
It should continue if either is found, which is what it does right now. If we seperate the check, then there's no way to failfast on no tree for a given stock. Plus this means we only have to lock once to grab it.